Source/IntegrationTests/TestFiles/LitTests/LitTest/reveal/revealInExpression.dfy

```diff
@@ -0,0 +1,39 @@
+// echo ''
+// NORUN: ! %verify --type-system-refresh --allow-axioms --bprint:%t.bpl --isolate-assertions --boogie "/printPruned:%S/pruned" %s > %t
+// NORUN: %diff "%s.expect" "%t"
+
+function P(x: int): bool {
+  true
+}
+
+function Q(x: int): bool 
+  requires P(0) {
+  true
+}
+
+method RevealExpressionScope()
+{
+  hide *;
+  var a := (reveal P; assert P(0); true);
+  assert P(1); // error
```

Comment:
Thanks. The change that accomplishes this puts the wellformedness check of a statement expression in a check-and-forget if block, so it on longer affects what comes after.

<img width="1020" alt="image" src="https://github.com/user-attachments/assets/36369a7f-48b6-461f-9ee6-c54e22a01ea3">

However, in a follow-up PR, we should make it so that all well-formedness checks are placed in check-and-forget blocks, to prevent "assert contamination" (the proof of one assert affecting the proof of other)

Reply:
Actually, I had to revert the above, and the test `revealInExpression` is disabled. I'll fix this in an upcoming PR: https://github.com/dafny-lang/dafny/pull/5626